test/plots/plotTests.ts

```diff
@@ -1,584 +1,302 @@
+        assert.strictEqual(plot.attr("foo", numberAccessor), plot, "setting the attribute returns the calling plot");
+
+        let attrAccessor = plot.attr("foo").accessor;
+        plot.datasets().forEach((dataset, datasetIndex) => {
+          dataset.data().forEach((datum, index) => {
+            assert.strictEqual(attrAccessor(datum, index, dataset), numberAccessor(datum, index),
+              `can set attribute for number datum ${index} in dataset ${datasetIndex}`);
+          });
+        });
+
+        const stringAccessor = (d: any, i: number) => `${d + i * 10} foo`;
+        assert.strictEqual(plot.attr("foo", stringAccessor), plot, "setting the attribute returns the calling plot");
+
+        attrAccessor = plot.attr("foo").accessor;
+        plot.datasets().forEach((dataset, datasetIndex) => {
+          dataset.data().forEach((datum, index) => {
+            assert.strictEqual(attrAccessor(datum, index, dataset), stringAccessor(datum, index),
+              `can set attribute for string datum ${index} in dataset ${datasetIndex}`);
+          });
+        });
+      });
+
+      it("can set the attribute based on the scaled input data", () => {
+        const data = [1, 2, 3, 4, 5];
+        const dataset = new Plottable.Dataset(data);
+        plot.addDataset(dataset);
+
+        const numberAccessor = (d: any, i: number) => d + i * 10;
+        const linearScale = new Plottable.Scales.Linear();
+        assert.strictEqual(plot.attr("foo", numberAccessor, linearScale), plot, "setting the attribute returns the calling plot");
+
+        let attrAccessor = plot.attr("foo").accessor;
+        let attrScale = plot.attr("foo").scale;
+        plot.datasets().forEach((dataset, datasetIndex) => {
+          dataset.data().forEach((datum, index) => {
+            assert.strictEqual(attrScale.scale(attrAccessor(datum, index, dataset)), linearScale.scale(numberAccessor(datum, index)),
+              `can set based on scaled version of number datum ${index} in dataset ${datasetIndex}`);
+          });
+        });
+
+        const stringAccessor = (d: any, i: number) => `${d + i * 10} foo`;
+        const categoryScale = new Plottable.Scales.Category();
+        categoryScale.domain(data.map(stringAccessor));
+        assert.strictEqual(plot.attr("foo", stringAccessor, categoryScale), plot, "setting the attribute returns the calling plot");
+
+        attrAccessor = plot.attr("foo").accessor;
+        attrScale = plot.attr("foo").scale;
+        plot.datasets().forEach((dataset, datasetIndex) => {
+          dataset.data().forEach((datum, index) => {
+            assert.strictEqual(attrScale.scale(attrAccessor(datum, index, dataset)), categoryScale.scale(stringAccessor(datum, index)),
+              `can set based on scaled version of string datum ${index} in dataset ${datasetIndex}`);
+          });
+        });
+      });
+
+      it("updates the scales extents when an attribute is set", () => {
```

Comment:
Might want to specify that it updates the extent of the `Scale` associated with that attribute.